### PR TITLE
Switch non-control-character validator to a trait

### PR DESCRIPTION
**Pull request by kyrias**

### CODE REVIEW

File: validator/src/lib.rs [hunk 1]
```diff
@@ -75,7 +75,9 @@ pub use validation::ip::{validate_ip, validate_ip_v4, validate_ip_v6};
 pub use validation::length::{validate_length, ValidateLength};
 pub use validation::must_match::validate_must_match;
 #[cfg(feature = "unic")]
-pub use validation::non_control_character::validate_non_control_character;
+pub use validation::non_control_character::{
```
Comment (Keats) [inline validator/src/lib.rs:78]: We could potentially make the trait not be gated behind the feature but then the default impl would be invalid... Not sure it's worth ti

**Review by Keats — CHANGES_REQUESTED** [review]
Some legit CI errors it seems

**Comment by kyrias** [comment]
> Some legit CI errors it seems

No, it's just that removing the assert function could only be done with the previous two PRs merged, since GitHub doesn't allow for stacking PRs across repos.

File: validator/src/lib.rs (hunk 1 above)
Comment (kyrias) [reply validator/src/lib.rs:78]: Hmm, why?

Comment (kyrias) [reply validator/src/lib.rs:78]: I guess I could see that, though it's a bit strange to have a default function implementation be conditionally compiled, and would also need the blanket implementation to be conditionally compiled.

Comment (Keats) [reply validator/src/lib.rs:78]: I agree. Let's not bother with it

**Pull request merged**